multimodal/src/autogluon/multimodal/models/mmocr_text_detection.py

```diff
@@ -0,0 +1,173 @@
+import logging
+import warnings
+from typing import Optional
+
+import torch
+from mim.commands.download import download
+from torch import nn
+
+try:
+    import mmcv
+    from mmcv.parallel import scatter
+    from mmcv.runner import load_checkpoint
+except ImportError:
+    mmcv = None
+
+try:
+    import mmocr
+    from mmocr.models import build_detector
+    from mmocr.utils.model import revert_sync_batchnorm
+except ImportError:
+    mmocr = None
+
+try:
+    import mmdet
+    from mmdet.core import get_classes
+except ImportError:
+    mmdet = None
+
+from ..constants import AUTOMM, BBOX, COLUMN, COLUMN_FEATURES, FEATURES, IMAGE, IMAGE_VALID_NUM, LABEL, LOGITS, MASKS
+from .utils import assign_layer_ids, get_column_features, get_model_head
+
+logger = logging.getLogger(AUTOMM)
+
+
+class MMOCRAutoModelForTextDetection(nn.Module):
+    """
+    Support MMOCR object detection models.
+    Refer to https://github.com/open-mmlab/mmdetection
+    """
+
+    def __init__(
+        self,
+        prefix: str,
+        checkpoint_name: str,
+        num_classes: Optional[int] = None,
+        pretrained: Optional[bool] = True,
+    ):
+        """
+        Load a pretrained object detector from MMdetection.
+
+        Parameters
+        ----------
+        prefix
+            The prefix of the MMdetAutoModelForObjectDetection model.
+        checkpoint_name
+            Name of the mmdet checkpoint.
+        num_classes
+            The number of classes.
+        pretrained
+            Whether using the pretrained mmdet models. If pretrained=True, download the pretrained model.
+        """
+        super().__init__()
+        logger.debug(f"initializing {checkpoint_name}")
+        self.checkpoint_name = checkpoint_name
+        self.pretrained = pretrained
+
+        # download config and checkpoint files using openmim
+        checkpoints = download(package="mmocr", configs=[checkpoint_name], dest_root=".")
+
+        # read config files
+        assert mmcv is not None, "Please install mmcv-full by: mim install mmcv-full."
+        config_file = checkpoint_name + ".py"
+        if isinstance(config_file, str):
+            self.config = mmcv.Config.fromfile(config_file)
+
+        # build model and load pretrained weights
+        assert mmocr is not None, "Please install MMOCR by: pip install mmocr."
+        # previous version1
```

Comment:
Please remove the commented lines if they are used for debugging.

Reply:
OK.